app/src/main/java/com/google/android/samples/socialite/ui/home/timeline/TimelineViewModel.kt

```diff
@@ -105,30 +123,71 @@ class TimelineViewModel @Inject constructor(
                 it.repeatMode = ExoPlayer.REPEAT_MODE_ONE
                 it.playWhenReady = true
                 it.addListener(videoSizeListener)
+                it.addListener(firstFrameListener)
             }
 
         videoRatio = null
         player = newPlayer
+
+        if (enablePreloadManager) {
+            initPreloadManager(loadControl)
+        }
+    }
+
+    private fun initPreloadManager(loadControl: DefaultLoadControl) {
+        preloadManager =
+            PreloadManagerWrapper.build(
+                (player as ExoPlayer).applicationLooper,
```

Comment:
`ExoPlayer.applicationLooper` and `ExoPlayer.playbackLooper` are different.

* [`applicationLooper`](https://github.com/androidx/media/blob/release/libraries/common/src/main/java/androidx/media3/common/Player.java#L2128) is associated with the application thread that's used to access the player from outside;
* [`playbackLooper`](https://github.com/androidx/media/blob/release/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/ExoPlayer.java#L1472) is associated with the playback thread; Internal playback operations are running on the playback thread.

`DefaultPreloadManager`'s preload looper should be the `ExoPlayer`'s playback looper.

Also, when initializing the `ExoPlayer`, if we do not inject a `playbackLooper` via `ExoPlayer.Builder.setPlaybackLooper`, then in the constructor of `ExoPlayer`, it will create an [`internalPlaybackThread`](https://github.com/androidx/media/blob/release/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/ExoPlayerImplInternal.java#L317) itself, and when the `ExoPlayer` instance is released, that `internalPlaybackThread` will be [quitted](https://github.com/androidx/media/blob/release/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/ExoPlayerImplInternal.java#L1545) by the player instance. This will cause issue if the looper associated with this thread is shared with `DefaultPreloadManager`. Imagine that when the `DefaultPreloadManager` is still doing something, and then looper quits and cannot process more messages. In this project, the issue is not that obvious because we release the preload manager and player together. 

On the other hand, if this `playbackLooper` is injected from the `ExoPlayer.Builder`, then the player instance won't quit it upon releasing.

So could we create a `HandlerThread` before initialising both, and pass the its looper to the `ExoPlayer.Builder` and the constructor of the `DefaultPreloadManager`? And after releasing both, we quit the thread in the app code?